### PR TITLE
[APR] getRateAtBlock for other chains

### PR DESCRIPTION
**Pull request by luizakp**
This PR adds getRateAtBlock for other chains.
This affects especially pools with `stMatic`.  `MaticX` case will be handled on BAL-781

### CODE REVIEW

File: apps/balancer-tools/src/app/apr/(utils)/chainsPublicClients.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  | typeof base;
+
+export type ChainName =
+  | "ethereum"
+  | "polygon"
+  | "polygon-zkevm"
+  | "arbitrum"
+  | "avalanche"
+  | "gnosis"
+  | "base";
```
Comment (ribeirojose) [inline apps/balancer-tools/src/app/apr/(utils)/chainsPublicClients.ts:28]: couldn't we use the enums we already have? :/

File: apps/balancer-tools/src/app/apr/(utils)/poolsRateProvider.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  },
+  {
+    poolAddress:
+      "0x577f6076e558818a5df21ce4acde9a9623ec0b4c000200000000000000000a64",
+    address: "0xeE652bbF72689AA59F0B8F981c9c90e2A8Af8d8f",
+    token: {
+      address: "0xfa68FB4628DFF1028CFEc22b4162FCcd0d45efb6",
+      symbol: "MaticX",
+    },
+    //TODO on BAL-781, MaticX does not update the rate daily.
```
Comment (ribeirojose) [inline apps/balancer-tools/src/app/apr/(utils)/poolsRateProvider.ts:19]: 🤔 interesting

File: apps/balancer-tools/src/app/apr/(utils)/tokenYield.ts [hunk 3]
```diff
@@ -81,37 +83,53 @@ function getAPRFromRate(
...
   return APR < 0 ? 0 : APR;
 }
 
 async function getPoolTokensRateProviders(chain: string, poolId: Address) {
   const data = await pools.gql(String(chain)).PoolRateProviders({ poolId });
 
-  if (!data.pool?.priceRateProviders?.length)
+  if (!data.pool?.priceRateProviders?.length) {
+    const poolRateProvider = manualPoolsRateProvider.find(
+      ({ poolAddress }) => poolAddress === poolId,
```
Comment (ribeirojose) [inline apps/balancer-tools/src/app/apr/(utils)/tokenYield.ts:94]:
```suggestion
      ({ poolAddress }) => poolAddress.toLowerCase() === poolId.toLowerCase(),
```

File: apps/balancer-tools/src/app/apr/(utils)/tokenYield.ts [hunk 4]
```diff
@@ -81,37 +83,53 @@ function getAPRFromRate(
...
   }
-
   return APR < 0 ? 0 : APR;
 }
 
 async function getPoolTokensRateProviders(chain: string, poolId: Address) {
   const data = await pools.gql(String(chain)).PoolRateProviders({ poolId });
 
-  if (!data.pool?.priceRateProviders?.length)
+  if (!data.pool?.priceRateProviders?.length) {
```
Comment (ribeirojose) [inline apps/balancer-tools/src/app/apr/(utils)/tokenYield.ts:92]: would we eventually use the manual approach to override bogus return values as well ? If so, we'd probably need to remove this if statement, right?

**Review by ribeirojose — DISMISSED** [review]
👍 lgtm, but would try to avoid adding yet another pattern for chain naming/mapping 😅

File: apps/balancer-tools/src/app/apr/(utils)/chainsPublicClients.ts (hunk 1 above)
Comment (luizakp) [reply apps/balancer-tools/src/app/apr/(utils)/chainsPublicClients.ts:28]: I tried but I got so complicated with the types that I decided to use this to make it work

**Comment by luizakp** [comment]
This is only working for a single pool like: https://tools-monorepo-ro0533nq7-bleu-fi.vercel.app/apr/pool/polygon/0xeab6455f8a99390b941a33bbdaf615abdf93455e000200000000000000000a66

**Pull request merged**